main.go

```diff
@@ -126,66 +124,77 @@ func main() {
-		return &nodeLocalController{}
-	case SystemController:
-		return &systemController{}
+func newDataMovementController(controller string) dataMovementControllerInterface {
+	switch controller {
+	case ManagerController:
+		return &managerController{}
+	case DefaultController:
+		return &defaultController{}
+	case NodeController:
+		return &nodeController{}
 	}
 
-	setupLog.Info("unable to create controller", "controller", "RsyncNodeDataMovement")
+	setupLog.Info("unable to create controller", "controller", controller)
 	os.Exit(1)
 	return nil
 }
 
-type nodeLocalController struct{}
+// Manager Controller is that which manages various data movement systems - Secrets, Deployments, and DaemonSets
+type managerController struct{}
 
-func (*nodeLocalController) GetType() string      { return NodeLocalController }
-func (*nodeLocalController) GetNamespace() string { return os.Getenv("NNF_NODE_NAME") }
+func (*managerController) GetType() string { return ManagerController }
+func (*managerController) SetOptions(opts *ctrl.Options) {
+	opts.Namespace = dmv1alpha1.DataMovementNamespace
+}
 
-func (*nodeLocalController) SetupReconcilers(mgr manager.Manager) (err error) {
-	if err = (&controllers.RsyncNodeDataMovementReconciler{
+func (c *managerController) SetupReconcilers(mgr manager.Manager) (err error) {
+	if err = (&controllers.DataMovementManagerReconciler{
 		Client: mgr.GetClient(),
 		Scheme: mgr.GetScheme(),
 	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "RsyncNodeDataMovement")
+		setupLog.Error(err, "unable to create controller", "controller", c.GetType())
 		os.Exit(1)
 	}
 
 	return
 }
 
-type systemController struct{}
+// Default Controller is that which operates on Data Movement resources.
+type defaultController struct{}
 
-func (*systemController) GetType() string      { return SystemController }
-func (*systemController) GetNamespace() string { return "" }
+func (*defaultController) GetType() string               { return DefaultController }
+func (*defaultController) SetOptions(opts *ctrl.Options) {}
 
-func (*systemController) SetupReconcilers(mgr manager.Manager) (err error) {
+func (c *defaultController) SetupReconcilers(mgr manager.Manager) (err error) {
 	if err = (&controllers.DataMovementReconciler{
 		Client: mgr.GetClient(),
 		Scheme: mgr.GetScheme(),
 	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "DataMovement")
-		os.Exit(1)
-	}
-
-	if err = (&controllers.RsyncTemplateReconciler{
-		Client: mgr.GetClient(),
-		Scheme: mgr.GetScheme(),
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "RsyncTemplate")
+		setupLog.Error(err, "unable to create controller", "controller", c.GetType())
 		os.Exit(1)
 	}
 
 	return
 }
+
+// Node controller is that which operates on Data Movement resources to particular NNF Nodes
+type nodeController struct {
```

Comment:
I do not see a user for this.

Reply:
line 142

Reply:
Okay, I don't see a "/manager --controller=node" anywhere...